File: flamingo-demo/src/main/java/org/pushingpixels/demo/flamingo/common/LocaleSwitcher.java
package org.pushingpixels.demo.flamingo.common;

import java.awt.event.ActionEvent;
import java.util.Locale;

import javax.swing.JComboBox;
import javax.swing.SwingUtilities;

public class LocaleSwitcher {
    static class LocaleInfo {
        Locale locale;

        String displayName;

        public LocaleInfo(Locale locale, String displayName) {
            this.locale = locale;
            this.displayName = displayName;
        }

        @Override
        public String toString() {
            return displayName;
        }
    }

    public interface LocaleCallback {
        public void onLocaleSelected(Locale selected);
    }

    public static JComboBox getLocaleSwitcher(final LocaleCallback callback) {
        Object[] locales = new Object[] { new LocaleInfo(new Locale("en", "US"), "English"),
                        new LocaleInfo(new Locale("iw", "IL"), "Hebrew") };
        final JComboBox result = new JComboBox(locales);
        result.setSelectedIndex(0);
        result.addActionListener((ActionEvent e) -> SwingUtilities.invokeLater(() -> {
            LocaleInfo selected = (LocaleInfo) result.getSelectedItem();
            callback.onLocaleSelected(selected.locale);
        }));
        return result;
    }
}

File: demos/plasma-demo/src/main/kotlin/org/pushingpixels/demo/plasma/LocaleSwitcher.kt
/*
 * Copyright (c) 2005-2020 Radiance Kirill Grouchnikov. All Rights Reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 *  o Redistributions of source code must retain the above copyright notice,
 *    this list of conditions and the following disclaimer.
 *
 *  o Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 *
 *  o Neither the name of the copyright holder nor the names of
 *    its contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
 * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
 * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
 * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
package org.pushingpixels.demo.plasma;

import org.pushingpixels.meteor.addDelayedActionListener
import java.util.*
import javax.swing.JComboBox

object LocaleSwitcher {
    data class LocaleInfo(val locale: Locale, val displayName: String) {
        override fun toString(): String {
            return displayName
        }
    }

    fun getLocaleSwitcher(callback: (Locale) -> Unit): JComboBox<LocaleInfo> {
        val locales = arrayOf(LocaleInfo(Locale("en", "US"), "English"),
                LocaleInfo(Locale("iw", "IL"), "Hebrew"))
        val result = JComboBox<LocaleInfo>(locales)
        result.selectedIndex = 0
        result.addDelayedActionListener {
            callback(result.model.getElementAt(result.selectedIndex).locale)
        }
        return result
    }
}
